theano/compile/ops.py

```diff
@@ -675,6 +675,8 @@ def __str__(self):
     def make_node(self, x):
         if self.axis.keys() and (x.ndim <= max(self.axis.keys())):
             raise ValueError('Trying to rebroadcast non-existent dimension')
+        temp_dict = OrderedDict([(axis + x.ndim, broad) if axis < 0 else (axis, broad) for axis, broad in self.axis.items()])
+        self.axis = temp_dict
```

Comment:
I mean, after the correction, you can still have negative axis. For example, if you have a matrix (2d) and use axis = -3. After correction, axis will be -1, which is not valid.